proposals/lip-0051.md

```diff
@@ -2447,7 +2448,7 @@ Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets fo
   * The `supplySubstore` array must be in ascending order of `tokenID`.
   * Across all elements of the `escrowSubstore` array, the `tokenID` must be unique (it appears at most once).
   * The `escrowSubstore` array must be in ascending order of `tokenID`.
-  * The `supportedTokensSubstore` array must be in lexicographical order of `chainID`.  For each entry of this array, the `supportedTokenIDs` array should be in lexicographic order.
+  * The `supportedTokensSubstore` array must be in lexicographical order of `chainID` and the `chainID` should be of length `CHAIN_ID_LENGTH` or `0`. For each entry of this array, the `supportedTokenIDs` array should be in lexicographic order.
```

Comment:
```suggestion
  * The `supportedTokensSubstore` array must be in lexicographical order of `chainID`. For each entry of this array, the `chainID` property should be of length `CHAIN_ID_LENGTH` or `0` and the `supportedTokenIDs` array should be in lexicographic order.
```

Reply:
Also, I think we should add the constraint that if there is an entry of length `0`, then it is the only entry of the array (since this implies that all tokens are supported, therefore does not make sense to have other entries).

Reply:
Agreed, thanks for this suggestion.

Reply:
Please see my comment below.